docs(mapValues): Improve return value description across all languages

wo-o29 opened the pull request:
## Summary  
  
This PR improves the return value description for the `mapValues` function across all language documentation (English, Korean, Japanese, and Simplified Chinese). The Korean documentation previously had the return value description in English, and all language versions now provide more specific descriptions that explicitly mention the `getNewValue` function.  
  
## Changes  
  
- Updated Korean documentation to use Korean language for the return value description  
- Enhanced all language versions to explicitly reference the `getNewValue` function in the return value description  
- Made descriptions more specific and clear about how values are transformed

wo-o29 committed:
docs(mapValues): improve return value description across all languages

wo-o29 committed:
docs(mapValues): change the section title to '인터페이스'

wo-o29 committed:
docs(mapValues): change section titles to Japanese for consistency

raon0211 commented:
We actually have a revised version of the es-toolkit docs now. Can you take a look at the new docs, see if any updates are needed, and open a new pull request if necessary?

wo-o29 commented:
> We actually have a revised version of the es-toolkit docs now. Can you take a look at the new docs, see if any updates are needed, and open a new pull request if necessary?

The new version of the document seems to have addressed all the issues I was looking for!🙃

Thank you 👍👍👍
